lightly/embedding/embedding.py

```diff
@@ -107,20 +107,23 @@ def embed(self,
         embeddings, labels, fnames = None, None, []
 
         if lightly._is_prefetch_generator_available():
-            pbar = tqdm(
-                BackgroundGenerator(dataloader, max_prefetch=3),
-                total=len(dataloader)
-            )
-        else:
-            pbar = tqdm(dataloader, total=len(dataloader))
+            dataloader = BackgroundGenerator(dataloader, max_prefetch=3),
+        
+        pbar = tqdm(
+            dataloader,
```

Comment:
You don't need to provide the dataloader as argument, see the example here:
https://github.com/tqdm/tqdm#manual